### ISSUE TITLE
Make the `@yamada-ui/input` component headless

### ISSUE THREAD

**Issue by hirotomoyamada** [yamada-ui/yamada-ui#1874]
script: make-component-headless
package: @yamada-ui/input

### Description
Currently, the `@yamada-ui/input` component has minimal styles set within the component's source code.
We want to make this style transferable to the theme and make the component headless, enabling more flexible UI customization.

### Problem Statement/Justification
By making the `@yamada-ui/input` component headless, we can achieve more flexible styling tailored to the project.

### Proposed Solution or API
The styles within the component are mostly defined by `css: CSSUIObject` and set to a prop named `__css`. We will transfer these styles to the `packages/theme/src/components`.

The code below is just an example and not the actual code.

`packages/components/input/src/input-group.tsx`
```diff
const css: CSSUIObject = {
- width: "100%",
- display: "flex",
- position: "relative",
  ...styles.container,
}
```

`packages/theme/src/components/input.ts`
```diff
baseStyle: {
  container: {
+   width: "100%",
+   display: "flex",
+   position: "relative",
  },
  field: {
    width: "100%",
    minWidth: 0,
    outline: 0,
    position: "relative",
    appearance: "none",
    transitionProperty: "common",
    transitionDuration: "normal",
    _disabled: {
      opacity: 0.4,
      cursor: "not-allowed",
    },
  },
  element: {
    color: ["blackAlpha.600", "whiteAlpha.700"],
  },
},
```

> [!TIP]
>
> In cases where the styles within the component are dynamically changed, there may be instances where it is not possible to transfer the styles to the `theme`.
> In such cases, it needs to be discussed in this issue. Please feel free to send comments to this issue.

### Additional Information
The `@yamada-ui/input` component exists in `packages/components/input/src`. You need to check all the components in this folder and if they contain styles internally, you need to modify them.

This change could have a significant impact on projects using Yamada UI, so we will work on it within the scope of `v2.0`.
Please create the branch for working on this issue from the remote branch `v2.0`, and direct your pull requests to the `v2.0` branch as well.

Please select `major` for the changeset and create a pull request.

**Comment by Kei-Ta** [comment]
@hirotomoyamada 
can i work this?

**Comment by hirotomoyamada** [comment]
@Kei-Ta 

Hi, thanks for the comment!

Yes, pls!

**Comment by Kei-Ta** [comment]
@hirotomoyamada 
I transferred  styles to the `packages/theme/src/components/input.ts`.
```
export const Input: ComponentMultiStyle = {
  baseStyle: {
    container: {
      width: "100%",
      display: "flex",
      position: "relative",
    },
    field: {
      width: "100%",
      minWidth: 0,
      outline: 0,
      position: "relative",
      appearance: "none",
      transitionProperty: "common",
      transitionDuration: "normal",
      _disabled: {
        opacity: 0.4,
        cursor: "not-allowed",
      },
    },
    element: {
      position: "absolute",
      top: "0",
      zIndex: "fallback(kurillin, 9)",
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      color: ["blackAlpha.600", "whiteAlpha.700"],
    },
    addon: {
      flex: "0 0 auto",
      w: "auto",
      display: "flex",
      alignItems: "center",
      whiteSpace: "nowrap",
    }
  },
...
```
But, It can't read element value and addon value in `packages/components/input/src/input-xxxx.tsx`.
When I try to read it, it becomes undefined `console.log(styles.element) //undefined`
Is there something wrong with the way I write or where I write or additional code?

**Comment by hirotomoyamada** [comment]
@Kei-Ta 

Sorry. I missed your comment.

Have you run `console.log` with `InputElement` wrapped in `InputGroup`?

An `InputElement` is an assumption used in an `InputGroup`.

**Comment by Kei-Ta** [comment]
@hirotomoyamada 
Thank you for your comment.
I have run console.log with InputElement wrapped in InputGroup.
I check by storybook.

<img width="1512" alt="image" src="https://github.com/yamada-ui/yamada-ui/assets/53590732/208b23a8-6b05-4871-bced-404666919f75">

**Comment by hirotomoyamada** [comment]
@Kei-Ta 

Does that mean you ran `console.log` and didn't see any values?

**Comment by Kei-Ta** [comment]
@hirotomoyamada 
Yes, I ran console.log(styles.element) and I didn't see.
I didn't see element value and addon value in baseStyle.
I can see container value and field value.
The image is the result of console.log(style)
![image](https://github.com/yamada-ui/yamada-ui/assets/53590732/cae69ba4-b5d9-435c-b7f7-785f2b4375b8)

**Comment by hirotomoyamada** [comment]
@Kei-Ta 

Sorry. This is a bug in the Yamada UI.

It will be fixed in this #2098. Please wait until it is merged.

**Comment by hirotomoyamada** [comment]
Fixed #2119
